resources/idlharness.js

```diff
@@ -916,14 +935,32 @@ IdlArray.prototype.collapse_partials = function()
                 this.members[parsed_idl.name].extAttrs.push(extAttr);
             }.bind(this));
         }
-        parsed_idl.members.forEach(function(member)
-        {
-            this.members[parsed_idl.name].members.push(new IdlInterfaceMember(member));
-        }.bind(this));
+        if (parsed_idl.members.length) {
+            test(function () {
+                parsed_idl.members.forEach(function(member)
+                {
+                    assert_true(
+                        this.members[parsed_idl.name].members.every(m => this.are_different_members(m, member)),
+                        "member " + member.name + " is unique");
+                    this.members[parsed_idl.name].members.push(new IdlInterfaceMember(member));
+                }.bind(this));
+            }.bind(this), `Partial ${parsed_idl.type} ${partialTestName}: member names are unique`);
+        }
     }.bind(this));
     this.partials = [];
 }
 
+IdlArray.prototype.are_different_members = function(m1, m2) {
```

Comment:
May I suggest inverting this to `are_duplicate_members` to make it clear even without the call sites that this is about detecting dupes, a case that should never happen? Other code that might care to compare members could just use `m1 === m2`, and shouldn't turn to this helper.

Reply:
Done.